core/controllers/email_dashboard.py

```diff
@@ -157,3 +161,40 @@ def get(self):
         }
 
         self.render_json(data)
+
+
+class EmailDashboardResult(base.BaseHandler):
+    """Handler for email dashboard result page."""
+    @require_valid_sender
+    def get(self, query_id):
+        query_model = user_models.UserQueryModel.get(query_id, strict=False)
+        if (query_model is None or
+                query_model.query_status != feconf.USER_QUERY_STATUS_COMPLETED
+                or query_model.submitter_id != self.user_id):
+            raise self.InvalidInputException('400 Invalid query id.')
+
+        self.values.update({
+            'query_id': query_id,
```

Comment:
I don't know how we can get `query_id` this way. `query_id` is embedded in URL so only handler for that URL has access to it. Even if we do send request to some other handler, how that handler will be able to identify which query's id we want ?
It can be extracted from URL directly, if Jinja way is not possible.